sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -346,6 +346,42 @@ object RapidsPluginUtils extends Logging {
       loadExtensions(classOf[SparkPlugin], pluginClasses)
     }
   }
+
+  /**
+   * Extracts supported GPU architectures from the given properties file
+   */
+  private def getSupportedGpuArchitectures(propFileName: String): Set[Int] = {
+    val props = RapidsPluginUtils.loadProps(propFileName)
+    Option(props.getProperty("gpu_architectures"))
+      .getOrElse(throw new RuntimeException(s"GPU architectures not found in $propFileName"))
+      .split(";")
+      .map(_.toInt)
+      .toSet
+  }
+
+  /**
+   * Checks if the current GPU architecture is supported by the spark-rapids-jni
+   * and cuDF libraries.
+   */
+  def checkGpuArchitecture(): Unit = {
+    val supportedGpuArchsSet = getSupportedGpuArchitectures(JNI_PROPS_FILENAME)
+      .intersect(getSupportedGpuArchitectures(CUDF_PROPS_FILENAME))
+    val majorGpuArch = Cuda.getComputeCapabilityMajor
+    val minorGpuArch = Cuda.getComputeCapabilityMinor
+    // Check if the device's major architecture is supported.
+    val supportedMajorArchs = supportedGpuArchsSet.filter(ver => ver / 10 == majorGpuArch)
+    if (supportedMajorArchs.isEmpty) {
+      throw new RuntimeException(s"GPU architecture $majorGpuArch$minorGpuArch is unsupported, " +
+        s"supported architectures: ${supportedGpuArchsSet.toSeq.sorted.mkString(", ")}")
+    }
+    // Check if the device's minor architecture is supported.
+    if (!supportedMajorArchs.exists(ver => ver % 10 == minorGpuArch)) {
```

Comment:
This should check for both major and minor at the same time:
```suggestion
    if (!supportedMajorArchs.exists(ver => ver == minorGpuArch + majorGpuArch * 10)) {
```

Reply:
Would not this be redundant? We have already filtered architectures matching major gpu arch